### ISSUE TITLE
db lock and/or integrity problems

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc#1214]
This is technically a duplicate, but the related issues are various and have been closed, e.g.: #889 #959 #952 #388 #920 #935

It may be solved by already by cylc-6 (after #1213 is addressed anyway).

Anyhow, for the record, after months without problems in one of our operational suites still running 5.4.14 we just suffered two of these DB errors in two days.  There was no operator intervention in the suite prior to either of the events.  First a DB integrity error, then DB locked.  Maybe the integrity error resulted in an abort that left the DB locked.

Here's the tail end of the error log 1:
```
2014/11/03 20:04:43 WARNING - [mtsat_create_csv.2014110319] -Assuming non-reported outputs were completed:
mtsat_create_csv.2014110319 submitted
Database Integrity Error: columns name, cycle are not unique:
	Converting INSERT to INSERT OR REPLACE for:
	request: INSERT INTO task_states VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
	args: ['ext_get_streamq', '2014110400', '2014-10-23T05:37:28', None, 0, None, 1, None, None, None, 'waiting']
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/threading.py", line 532, in __bootstrap_inner
    self.run()
  File "/gpfs_oper/filesets/oper/admin/um_fcm/cylc/cylc-5.4.14/lib/cylc/rundb.py", line 155, in run
    raise Exception(self.generic_err_msg%(type(e),str(e),req,arg))
Exception: <class 'sqlite3.OperationalError'>:database is locked occurred while trying to run:
	request: INSERT OR REPLACE INTO task_states VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
	args: ['ext_get_streamq', '2014110400', '2014-10-23T05:37:28', None, 0, None, 1, None, None, None, 'waiting']

Traceback (most recent call last):
  File "/gpfs_oper/filesets/oper/admin/um_fcm/cylc/cylc-5.4.14/lib/cylc/run.py", line 76, in main
    server.run()
  File "/gpfs_oper/filesets/oper/admin/um_fcm/cylc/cylc-5.4.14/lib/cylc/scheduler.py", line 1106, in run
    raise self.db.c.exception
OperationalError: database is locked
ERROR CAUGHT: cleaning up before exit
2014/11/03 21:15:01 WARNING - some active tasks will be orphaned
THE ERROR WAS:
database is locked
```

And error log 2:
```
2014/11/05 02:04:46 WARNING - [mtsat_create_csv.2014110501] -Assuming non-reported outputs were completed:
mtsat_create_csv.2014110501 submitted
Database Integrity Error: columns name, cycle are not unique:
	Converting INSERT to INSERT OR REPLACE for:
	request: INSERT INTO task_states VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
	args: ['ext_get_clidbdata_synop', '2014110506', '2014-11-03T23:14:13', None, 0, None, 1, None, None, None, 'waiting']
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/threading.py", line 532, in __bootstrap_inner
    self.run()
  File "/gpfs_oper/filesets/oper/admin/um_fcm/cylc/cylc-5.4.14/lib/cylc/rundb.py", line 155, in run
    raise Exception(self.generic_err_msg%(type(e),str(e),req,arg))
Exception: <class 'sqlite3.OperationalError'>:database is locked occurred while trying to run:
	request: UPDATE task_states SET status=?, try_num=?, submit_num=?, time_updated=? WHERE name==? AND cycle==?
	args: ['running', 1, 1, '2014-11-05T03:00:52', 'mtsat_hrit_pc_ftp_cleanup', '2014110502']

Traceback (most recent call last):
  File "/gpfs_oper/filesets/oper/admin/um_fcm/cylc/cylc-5.4.14/lib/cylc/run.py", line 76, in main
    server.run()
  File "/gpfs_oper/filesets/oper/admin/um_fcm/cylc/cylc-5.4.14/lib/cylc/scheduler.py", line 1106, in run
    raise self.db.c.exception
OperationalError: database is locked
ERROR CAUGHT: cleaning up before exit
2014/11/05 03:04:53 WARNING - some active tasks will be orphaned
THE ERROR WAS:
database is locked
```

**Comment by hjoliver** [comment]
@arjclark - would you agree we can close, thanks to #1295.

**Comment by arjclark** [comment]
Closing for now as I think between #1233 and #1295 we've done all we can for this. Locking and recovery from locking of the viewable db to be addressed by #1213.

**Comment by hjoliver** [comment]
See also #1493.
